src/python/ensembl/io/genomio/seq_region/rename.py

```diff
@@ -0,0 +1,212 @@
+            seqr.operation = Operation.not_found
+            continue
+        for row in rows:
+            db_seqr: SeqRegion = row[0]
+            if not db_seqr:
+                seqr.operation = Operation.not_found
+            attribs = get_attribs_dict(db_seqr)
+            db_brc_name = attribs.get("BRC4_seq_region_name", "")
+            seqr.old_brc_name = db_brc_name
+            if not db_brc_name:
+                logging.info(f"Seq region {seqr.name} doesn't have a BRC name")
+                seqr.operation = Operation.insert
+                continue
+            if seqr.brc_name == db_brc_name:
+                logging.info(f"Seq region {seqr.name} already exists with same name")
+                seqr.operation = Operation.do_nothing
+                continue
+            logging.info(f"Seq region {seqr.name} already exists as {db_brc_name} instead of {seqr.brc_name}")
+            seqr.operation = Operation.update
+            seqr.seq_region_id = db_seqr.seq_region_id
+            seq_regions.append(seqr)
+
+    return seq_regions
+
+
+def get_attribs(seq_region: SeqRegion) -> List:
+    """Given a seq_region, extract the attribs as value-source items.
+
+    Args:
+        seq_region (SeqRegion): The seq_region from which the attribs are extracted.
+
+    Returns:
+        All attributes as a list of dictionaries with 'value' and 'source' keys.
+    """
+    attribs = seq_region.seq_region_attrib
+    atts = []
+    if attribs:
+        for attrib in attribs:
+            att_obj = {"value": attrib.value, "source": attrib.attrib_type.code}
+            atts.append(att_obj)
+    return atts
+
+
+def get_attribs_dict(seq_region: SeqRegion) -> Dict[str, Any]:
+    """Extracts all the attributes of the given sequence region.
+
+    Args:
+        seq_region: Sequence region.
+
+    Returns:
+        Pairs of source and value for each attribute.
+    """
+
+    attribs = get_attribs(seq_region)
+    attrib_dict = {attrib["source"]: attrib["value"] for attrib in attribs}
+    return attrib_dict
+
+
+def update_seq_region_name(
+    session: DBConnection, seq_region: SeqRegionReplacement, update: bool = False
+) -> None:
+    """Update BRC seq_region based on the operation provided.
+
+    Args:
+        session: SQLAlchemy session.
+        seq_region: Seq region replacement.
+        update: Make actual changes to the database.
+
+    """
+    if not update:
+        logging.info(f"Rename {seq_region} (fake update)")
+        return
+    logging.info(f"Rename {seq_region}")
+
+    brc_attrib_id = 548
+    if seq_region.operation == Operation.update:
+        stmt = select(SeqRegionAttrib).where(
+            SeqRegionAttrib.seq_region_id == seq_region.seq_region_id,
+            SeqRegionAttrib.attrib_type_id == brc_attrib_id,
+        )
+        seq_attrib = session.scalars(stmt).one()
+        seq_attrib.value = seq_region.brc_name
+    elif seq_region.operation == Operation.insert:
```

Comment:
I am not able to replicate this case if i have such case as 
`if not db_brc_name:
                logging.info(f"Seq region {seqr.name} doesn't have a BRC name")`
it gives the log message however `seq_regions = []` so doesnt apply the insert operation for me

Reply:
Do you get the message that says the insert operation is not supported (I have only implemented the update part)?

Reply:
The value for `return seq_regions` in `get_seq_regions_to_replace` is an empty list for me, so I dont get any message